### PR TITLE
add recipe introducing servant-pagination

### PR DESCRIPTION
**Pull request by KtorZ**
As proposed in #125, here's a short introduction to [servant-pagination](https://hackage.haskell.org/package/servant-pagination-2.1.0). 

Feedback is welcome :)

### CODE REVIEW

**Comment by alpmestan** [comment]
@KtorZ Haven't yet looked at the explanations (I won't be too picky though, don't worry), but is there a reason why your patch comes with both a `Main.hs` and a `Pagination.lhs`, with the code duplicated in both?

See https://github.com/haskell-servant/servant/commit/5ac6de82773bcf2f352dda432a07225178feba2b for an example of commit that adds a cookbook recipe with the code and explanations in a single file. The trick is to have:

```
  ghc-options:         -pgmL markdown-unlit
  build-tool-depends: markdown-unlit:markdown-unlit
```

in the .cabal file. This processes the file with markdown-unlit (which strips away all the markdown, only leaving the code) before giving its contents to GHC.

**Comment by KtorZ** [comment]
Ah! You're entirely right, I wrongly committed that. I'll update :+1:

**Comment by KtorZ** [comment]
Done :+1: 
Let me know :)

File: doc/cookbook/pagination/Pagination.lhs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  getFieldValue _ = hex
+
+-- to then define: Ranges '["name", "hex"] Color
+```
+
+
+#### Parsing Options
+
+By default, `servant-pagination` provides an implementation of `getRangeOptions` for each
+`HasPagination` type-class. However, this can be overwritten when defining a instance of that
```
Comment (alpmestan) [inline doc/cookbook/pagination/Pagination.lhs]: for each `HasPagination` **instance**, I suppose?

Comment (KtorZ) [reply doc/cookbook/pagination/Pagination.lhs]: Indeed!

*hide in shame*

Comment (KtorZ) [reply doc/cookbook/pagination/Pagination.lhs]: Fixed in 93838ae9e76ffa18dcf4ff344b52decccd5a6db2

**Comment by alpmestan** [comment]
Thanks @KtorZ!

Your recipe should appear on https://haskell-servant.readthedocs.io/en/latest/cookbook/index.html (note the `latest` in the URL, which reflects the docs for the `master` branch, as opposed to `stable` which has the docs for the latest release) soon (15 mins IIRC? something along those lines).

**Comment by KtorZ** [comment]
Awesome! :tada:

**Comment by alpmestan** [comment]
Here it is: https://haskell-servant.readthedocs.io/en/latest/cookbook/pagination/Pagination.html :-)

**Comment by ysangkok** [comment]
@KtorZ the pagination library is missing a bunch of dependency bound updates, are you still maintaining this? I think it would be best if either the repo became part of this org, or that we removed it.

**Comment by KtorZ** [comment]
I am not directly, though it's part of the [Chordify](https://github.com/chordify/) organization which I am (unfortunately ❤️) no longer working with. 
I don't know if they're still keen on maintaining it but perhaps @jbransen can chime in.

**Comment by jbransen** [comment]
@isomorpheme ^

**Comment by isomorpheme** [comment]
Chiming in as one of the Chordify people with open source responsibility. :) If the dependency bounds can be fixed purely via a Hackage revision then @swamp-agr should be able to do that; if not I'd be happy to review a patch.

In general I expect we'll maintain it just to the degree that we need for internal use - we're open to PRs of course but I've noticed that reviewing those tends to drift down to the bottom of my todo list. So with that in mind I wouldn't mind at all to transfer the repo to this org (but removing the cookbook instead is also fine by me).

**Comment by ysangkok** [comment]
Patch submitted at haskell-servant-pagination/pull/24 , @isomorpheme

**Comment by ysangkok** [comment]
@isomorpheme Please review the linked patch, it seems like swamp-agr is not available.

**Pull request merged**